src/ui/QueryForCommerce/QueryForCommerce.ts

```diff
@@ -0,0 +1,71 @@
+import { Component } from '../Base/Component';
+import { IComponentBindings } from '../Base/ComponentBindings';
+import { ComponentOptions } from '../Base/ComponentOptions';
+import { Assert } from '../../misc/Assert';
+import { QueryEvents, IBuildingQueryEventArgs } from '../../events/QueryEvents';
+import { AnalyticsEvents, IChangeAnalyticsCustomDataEventArgs } from '../../events/AnalyticsEvents';
+import { Initialization } from '../Base/Initialization';
+import { exportGlobally } from '../../GlobalExports';
+
+export interface IQueryForCommerceOptions {
+  listing?: string;
+}
+
+/**
+ * The QueryForCommerce component enables to set properties for listing pages in commerce sites in a convenient way
```

Comment:
I would change the documentation to explain that it's a "generic" component that handles commerce specific attributes (and not only the listing)

@fbeaudoincoveo we can maybe discuss a bit more (live or slack) about the long term goal of the component so that we get a future proof description.